src/triggers/comment.ts

```diff
@@ -37,6 +80,34 @@ interface CommentsResponse {
 const getCommentList = () => async (z: ZObject, bundle: Bundle) => {
   const cursor = bundle.meta.page ? await z.cursor.get() : undefined;
 
+  const variables = {
+    creatorId: bundle.inputData.creator_id,
+    teamId: bundle.inputData.team_id,
+    issueId: bundle.inputData.issue,
+    projectUpdateProjectId: bundle.inputData.project_update_project_id,
+    after: cursor,
+  };
+
+  const issueTeamFilter = variables.teamId ? `{ issue: { team: { id: { eq: $teamId } } } }` : "";
+  const issueIdFilter = variables.issueId ? `{ issue: { id: { eq: $issueId } } }` : "";
+  const projectUpdateProjectFilter = variables.projectUpdateProjectId
+    ? `{ projectUpdate: { project: { id: { eq: $projectUpdateProjectId }}}}`
+    : "";
+  const creatorFilter = variables.creatorId ? `{ user: { id: { eq: $creatorId } } }` : "";
+
+  let filterString = "";
+
+  if (creatorFilter || issueTeamFilter || issueIdFilter || projectUpdateProjectFilter) {
+    filterString = `filter: {
+      and: [
+        ${creatorFilter}
+        ${issueTeamFilter}
+        ${issueIdFilter}
+        ${projectUpdateProjectFilter}
+      ]
+    }`;
+  }
```

Comment:
Why do you need to do this? Previous approach when always defining the filters with not set related variable should work right?

Reply:
As a matter of fact, they are mutually exclusive (based on the current implementation), so we need to apply them conditionally based on a current variables provided by a user